src/mappings/genesisProtocol/GenesisProtocol.ts

```diff
@@ -87,6 +90,19 @@ export function handleStake(event: Stake): void {
     return;
   }
 
+  let proposal = store.get(
```

Comment:
could you wrap this with a function (e.g stateUpdate) as it is called multiple times?